### DISCUSSION TITLE
Cannot create a Graph_asset with Partition

### DISCUSSION THREAD

**Discussion by dridk** [dagster-io/dagster#20188] [Q&A] [answered]
Hi,

I m trying to create a partition assets with different step. I m using graph_asset for this purpose. 

I cannot read context.asset_partition_key_for_output()  within op .   It returns :

 ```
 dagster._check.CheckError: Failure condition: The output has no asset partitions 
```

```python
@op 
def test1(context):
    context.log.debug(context.asset_partition_key_for_output()) # <===  ERROR 
    return 1 


@op 
def test2(context):
    return 2 

@op 
def merge(a,b):
     return a + b 


@graph_asset( partitions_def = df.MonthlyPartitionDefinition(start_date="2013-01-01"))
def myasset():
   a = test1()
   b = test2() 
   c = merge(a,b)
   return c







```

**Comment by OwenKephart** [comment, marked as answer]
Hi @dridk -- in this case, you can use `context.partition_key` instead. It's a bit confusing, but technically `test1` is not actually responsible for materializing any asset (only `merge` is), and so the asset-related methods there will not work.
